### ISSUE TITLE
sqlserver returning `None` result from cursor

### ISSUE THREAD

**Issue by conorbranagan** [DataDog/dd-agent#1553]
User case: https://datadog.zendesk.com/agent/tickets/24695

Traceback:
```
2015-04-15 09:27:41 Mountain Daylight Time | WARNING | checks.sqlserver(sqlserver.py:84) | Can't load the metric sqlserver.buffer.cache_hit_ratio, ignoring
Traceback (most recent call last):
  File "C:\Program Files (x86)\Datadog\Datadog Agent\files\..\checks.d\sqlserver.py", line 75, in __init__
  File "C:\Program Files (x86)\Datadog\Datadog Agent\files\..\checks.d\sqlserver.py", line 200, in get_sql_type
TypeError: 'NoneType' object is not iterable
```

Logs show that it happens after running for a while so I'm thinking that maybe the connection is getting closed. Another thing to consider is that `cursor.fetchone()` will return `None` when no data remains [see adodbapi docs](http://adodbapi.sourceforge.net/quick_reference.pdf)

**Comment by yannmh** [comment]
@LeoCavaille is it solved by #1631 ?

**Comment by yannmh** [comment]
It should. Closing
